### DISCUSSION TITLE
9. TCP의 패킷이 순서가 바뀌었을 때 어떻게 아나요?

### DISCUSSION THREAD

**Discussion by eun-byeol** [jewelry-box/cs-interview#14] [General]
TCP의 패킷이 순서가 바뀌었을 때 어떻게 아나요?

**[키워드]**
- TCP 세그먼트 헤더 구조
- 순서 번호 필드

**Comment by cutehumanS2** [comment]
TCP는 **순서 번호**(시퀀스 번호)를 사용하여 데이터의 순서를 추적합니다. 
이 번호는 각 패킷이 포함하는 데이터의 시작 위치를 나타내며, 수신 호스트는 도착한 패킷의 순서 번호를 확인하여 패킷의 순서가 올바른지 판단합니다. 
만약 순서가 어긋난 패킷이 도착하면, 수신 호스트는 이를 버퍼에 임시 저장하여 누락된 패킷이 도착할 때까지 기다린 뒤 재조립하거나, 누락된 패킷에 대한 중복 ACK를 송신하여 재전송을 요청합니다.
